// angular-msda/esm2015/lib/toast/toast.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Injectable, Injector, ComponentFactoryResolver, ApplicationRef } from '@angular/core';
import { MSDAToast } from './toast';
export class MSDAToastConfig {
    constructor() {
        this.onClose = () => { };
        this.linkClick = (e) => { };
        this.closeAfter = 3000;
    }
}
if (false) {
    /** @type {?} */
    MSDAToastConfig.prototype.onClose;
    /** @type {?} */
    MSDAToastConfig.prototype.linkClick;
    /** @type {?} */
    MSDAToastConfig.prototype.closeAfter;
}
;
export class MSDAToastService {
    /**
     * @param {?} componentFactoryResolver
     * @param {?} appRef
     * @param {?} injector
     */
    constructor(componentFactoryResolver, appRef, injector) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.appRef = appRef;
        this.injector = injector;
    }
    /**
     * @param {?} html
     * @param {?=} userConfig
     * @return {?}
     */
    showHtml(html, userConfig) {
        /** @type {?} */
        const config = Object.assign({}, new MSDAToastConfig(), userConfig);
        // 1. Create a component reference from the component
        this._componentRef = this.componentFactoryResolver
            .resolveComponentFactory(MSDAToast)
            .create(this.injector);
        this._componentRef.instance.html = html;
        this._componentRef.instance.linkClick.subscribe((e) => config.linkClick(e));
        this._componentRef.instance.dismiss = () => {
            this._componentRef.instance.linkClick.unsubscribe();
            this.appRef.detachView(this._componentRef.hostView);
            this._componentRef.destroy();
            config.onClose();
        };
        // 2. Attach component to the appRef so that it's inside the ng component tree
        this.appRef.attachView(this._componentRef.hostView);
        // 3. Get DOM element from component
        /** @type {?} */
        const domElem = (/** @type {?} */ (((/** @type {?} */ (this._componentRef.hostView)))
            .rootNodes[0]));
        // 4. Append DOM element to the body
        document.body.appendChild(domElem);
        // 5. Wait some time and remove it from the component tree and from the DOM
        setTimeout(() => this._componentRef.instance.dismiss(), config.closeAfter);
    }
}
MSDAToastService.decorators = [
    { type: Injectable }
];
/** @nocollapse */
MSDAToastService.ctorParameters = () => [
    { type: ComponentFactoryResolver },
    { type: ApplicationRef },
    { type: Injector }
];
if (false) {
    /** @type {?} */
    MSDAToastService.prototype._componentRef;
    /** @type {?} */
    MSDAToastService.prototype.componentFactoryResolver;
    /** @type {?} */
    MSDAToastService.prototype.appRef;
    /** @type {?} */
    MSDAToastService.prototype.injector;
}
