// angular-msda/esm5/lib/card-button/card-buttons-container.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Input, Output, Component, ViewEncapsulation, ChangeDetectionStrategy, EventEmitter, ElementRef } from '@angular/core';
var MSDACardButtonsContainer = /** @class */ (function () {
    function MSDACardButtonsContainer(_el) {
        this._el = _el;
        this.row = 3;
        this.column = 3;
        this.dragEnabled = false;
        this.dataSource = [];
        this.sortupdate = new EventEmitter();
        this.buttonClick = new EventEmitter();
    }
    Object.defineProperty(MSDACardButtonsContainer.prototype, "gridTemplateColumns", {
        get: /**
         * @return {?}
         */
        function () {
            return '1fr '.repeat(this.column).trim();
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    MSDACardButtonsContainer.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.dragEnabled) {
            jQuery('.msda-card-buttons-container-inner', this._el.nativeElement).sortable({
                handle: '.msda-card-button-move-handler'
            }).bind('sortupdate', function (event, ui) {
                _this.sortupdate.emit({ event: event, ui: ui });
            });
        }
    };
    MSDACardButtonsContainer.decorators = [
        { type: Component, args: [{
                    selector: 'msda-card-buttons-container',
                    exportAs: 'MSDACardButtonsContainer',
                    template: "<div class=\"msda-card-buttons-container-inner\" [style.grid-template-columns]=\"gridTemplateColumns\">\r\n  <ng-template ngFor let-item [ngForOf]=\"dataSource\">\r\n    <msda-card-button [title]=\"item.title\" [text]=\"item.text\" [icon]=\"item.icon\" [iconType]=\"item.iconType\" [iconBackgroundColor]=\"item.iconBackgroundColor\" (buttonClick)=\"buttonClick.emit($event)\"></msda-card-button>\r\n  </ng-template>\r\n</div>\r\n",
                    encapsulation: ViewEncapsulation.None,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    host: {
                        'role': 'button',
                        'class': 'msda-card-buttons-container'
                    },
                    styles: [".msda-card-buttons-container{overflow:hidden}.msda-card-buttons-container-inner{display:-ms-grid;display:grid;margin-top:10px;margin-bottom:10px;gap:10px 24px}"]
                }] }
    ];
    /** @nocollapse */
    MSDACardButtonsContainer.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    MSDACardButtonsContainer.propDecorators = {
        row: [{ type: Input }],
        column: [{ type: Input }],
        dragEnabled: [{ type: Input }],
        dataSource: [{ type: Input }],
        sortupdate: [{ type: Output }],
        buttonClick: [{ type: Output }]
    };
    return MSDACardButtonsContainer;
}());
export { MSDACardButtonsContainer };
if (false) {
    /** @type {?} */
    MSDACardButtonsContainer.prototype.row;
    /** @type {?} */
    MSDACardButtonsContainer.prototype.column;
    /** @type {?} */
    MSDACardButtonsContainer.prototype.dragEnabled;
    /** @type {?} */
    MSDACardButtonsContainer.prototype.dataSource;
    /** @type {?} */
    MSDACardButtonsContainer.prototype.sortupdate;
    /** @type {?} */
    MSDACardButtonsContainer.prototype.buttonClick;
    /** @type {?} */
    MSDACardButtonsContainer.prototype._el;
}
