// angular-msda/esm5/lib/toast/toast.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Injectable, Injector, ComponentFactoryResolver, ApplicationRef } from '@angular/core';
import { MSDAToast } from './toast';
var MSDAToastConfig = /** @class */ (function () {
    function MSDAToastConfig() {
        this.onClose = function () { };
        this.linkClick = function (e) { };
        this.closeAfter = 3000;
    }
    return MSDAToastConfig;
}());
export { MSDAToastConfig };
if (false) {
    /** @type {?} */
    MSDAToastConfig.prototype.onClose;
    /** @type {?} */
    MSDAToastConfig.prototype.linkClick;
    /** @type {?} */
    MSDAToastConfig.prototype.closeAfter;
}
;
var MSDAToastService = /** @class */ (function () {
    function MSDAToastService(componentFactoryResolver, appRef, injector) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.appRef = appRef;
        this.injector = injector;
    }
    /**
     * @param {?} html
     * @param {?=} userConfig
     * @return {?}
     */
    MSDAToastService.prototype.showHtml = /**
     * @param {?} html
     * @param {?=} userConfig
     * @return {?}
     */
    function (html, userConfig) {
        var _this = this;
        /** @type {?} */
        var config = tslib_1.__assign({}, new MSDAToastConfig(), userConfig);
        // 1. Create a component reference from the component
        this._componentRef = this.componentFactoryResolver
            .resolveComponentFactory(MSDAToast)
            .create(this.injector);
        this._componentRef.instance.html = html;
        this._componentRef.instance.linkClick.subscribe(function (e) { return config.linkClick(e); });
        this._componentRef.instance.dismiss = function () {
            _this._componentRef.instance.linkClick.unsubscribe();
            _this.appRef.detachView(_this._componentRef.hostView);
            _this._componentRef.destroy();
            config.onClose();
        };
        // 2. Attach component to the appRef so that it's inside the ng component tree
        this.appRef.attachView(this._componentRef.hostView);
        // 3. Get DOM element from component
        /** @type {?} */
        var domElem = (/** @type {?} */ (((/** @type {?} */ (this._componentRef.hostView)))
            .rootNodes[0]));
        // 4. Append DOM element to the body
        document.body.appendChild(domElem);
        // 5. Wait some time and remove it from the component tree and from the DOM
        setTimeout(function () { return _this._componentRef.instance.dismiss(); }, config.closeAfter);
    };
    MSDAToastService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    MSDAToastService.ctorParameters = function () { return [
        { type: ComponentFactoryResolver },
        { type: ApplicationRef },
        { type: Injector }
    ]; };
    return MSDAToastService;
}());
export { MSDAToastService };
if (false) {
    /** @type {?} */
    MSDAToastService.prototype._componentRef;
    /** @type {?} */
    MSDAToastService.prototype.componentFactoryResolver;
    /** @type {?} */
    MSDAToastService.prototype.appRef;
    /** @type {?} */
    MSDAToastService.prototype.injector;
}
